backend/databasemodel/model.sql

```diff
@@ -1543,6 +1582,30 @@ COMMENT ON COLUMN kooste.metadata.datasource IS E'You can get this information f
 ALTER TABLE kooste.metadata OWNER TO tarmo_admin;
 -- ddl-end --
 
+INSERT INTO kooste.metadata (
```

Comment:
Here, datasource string is missing, that's why all tests fail.

I don't know why kooste.metadata table is here, i.e. what is it used for? Anyway, if the table is needed, apparently datasource cannot be NULL.

Reply:
I see. Naturally, I could just set datasource to tarmo_admin or something like that. The idea of kooste.metadata table was to store the information about the updates in kooste schema. Actually, we should probably add also a field called updated_table to kooste.metadata in order to have all relevant information here (e.g. which table has been updated and from which datasource the data has been fetched and when the update has happened + id for updating event).

Reply:
Okay, now I'm confused about the kooste.metadata table as opposed to lipas.metadata table, kooste.osm_metadata table and other metadata tables.

At the moment, each individual metadata table contains a row that tells 1) what data should be imported and 2) when the data was last imported.

As it is, it's a bit confusing, since the row tells us what data was imported at last import and when, and the same row is used to tell that the same data should be imported again at next import. At next import, the date is just overwritten.

Is the intention that each metadata table should contain multiple rows, so it contains a list of all the imports that have been done? If so, why is the row also used to tell about *what* to import at next import?

And anyway, all metadata tables tell us when each dataset was updated to kooste. Should kooste.metadata table duplicate this data somehow?

Reply:
Huh, these metadata tables are causing me a headache too..

Maybe we should keep the same idea as with lipas.metadata table which only contains info about what data should be imported. This would mean that
- for kooste.osm_metadata we remove the update_id and last_modified field
- for kooste.metadata we remove the update_id, last_modified field and datasource
- we add a new table kooste.log_info with fields update_id, last_modified and datasource and use this table to store information about when each data has been imported. Each import creates a new row to this table. Can we populate the datasource field somehow automatically e.g. with tablename the data has been imported to?

Reply:
No, lipas_metadata table *also* contains last_modified field, like all the others. It is also used.

So, all the metadata tables work the same way: they tell the last modified time, and data to import, on a single row. Kooste.metadata is not used anywhere, because we don't import any data from kooste to kooste.

Reply:
Also, we *do* need the metadata tables for syke, museovirasto or tre wfs, and they are all used already at the moment. They contain exactly the names of the layers that should be imported.

All the importers work identically. They check the corresponding metadata table to see what data should be imported, and what data should be left out. I thought that is the idea too.

Reply:
So, currently *all* the other metadata tables are used for two purposes: saving the import time, and checking which data to import. Kooste.metadata is not used anywhere at all.

Reply:
Ups.

Well the same propose as above but by removing the update_id and last_modified field from lipas.metadata table too. So that in the metadata tables we only have info needed for importing and the actual events would be stored in combined kooste.log_info table? In other words the only purpose of kooste.metadata table would be to store tarmo_category jsonb lists.

Reply:
Well, now after we have discussed about this it would definately feel more clear to have separate tables for what should be imported (call these now metadata tables) and what was imported & when (call these log tables)